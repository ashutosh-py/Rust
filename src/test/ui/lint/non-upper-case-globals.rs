// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![allow(dead_code)]
#![deny(non_upper_case_globals)]

static foo: isize = 1;
//~^ ERROR static variable `foo` should have an upper case name such as `FOO`

static mut bar: isize = 1;
//~^ ERROR static variable `bar` should have an upper case name such as `BAR`

fn main() {
    const b: usize = 1;
    //~^ ERROR constant `b` should have an upper case name such as `B`
}

trait Foo {
    const camelCase: usize;
    //~^ ERROR associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
}

impl Foo for i32 {
    const camelCase: usize = 2;
    //~^ ERROR associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
}
