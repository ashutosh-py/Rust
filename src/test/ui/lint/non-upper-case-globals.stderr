error: static variable `foo` should have an upper case name such as `FOO`
  --> $DIR/non-upper-case-globals.rs:14:8
   |
14 | static foo: isize = 1;
   |        ^^^
   |
note: lint level defined here
  --> $DIR/non-upper-case-globals.rs:12:9
   |
12 | #![deny(non_upper_case_globals)]
   |         ^^^^^^^^^^^^^^^^^^^^^^

error: static variable `bar` should have an upper case name such as `BAR`
  --> $DIR/non-upper-case-globals.rs:17:12
   |
17 | static mut bar: isize = 1;
   |            ^^^

error: constant `b` should have an upper case name such as `B`
  --> $DIR/non-upper-case-globals.rs:21:11
   |
21 |     const b: usize = 1;
   |           ^

error: associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
  --> $DIR/non-upper-case-globals.rs:26:11
   |
26 |     const camelCase: usize;
   |           ^^^^^^^^^

error: associated constant `camelCase` should have an upper case name such as `CAMEL_CASE`
  --> $DIR/non-upper-case-globals.rs:31:11
   |
31 |     const camelCase: usize = 2;
   |           ^^^^^^^^^

error: aborting due to 5 previous errors

